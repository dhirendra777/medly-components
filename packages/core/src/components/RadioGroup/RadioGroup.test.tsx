import { cleanup, fireEvent, render } from '@test-utils';
import React from 'react';
import { RadioGroup } from './RadioGroup';

describe('Radio Group', () => {
    afterEach(cleanup);

    it('should render correctly', () => {
        const { container } = render(
            <RadioGroup
                disabled
                fullWidth
                label="Gender"
                name="gender"
                value="female"
                options={[
                    { value: 'female', label: 'Female' },
                    { value: 'male', label: 'Male' }
                ]}
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should call onChange prop with selected option', () => {
        const mockOnChange = jest.fn();

        const { container, getByText } = render(
            <RadioGroup
                label="Gender"
                name="gender"
                onChange={mockOnChange}
                value="female"
                required
                size="M"
                labelPosition="top"
                options={[
                    { value: 'female', label: 'Female' },
                    { value: 'male', label: 'Male' }
                ]}
            />
        );
        fireEvent.click(getByText('Male'));
        expect(mockOnChange).toBeCalledWith('male');
        expect(container).toMatchSnapshot();
    });
});
