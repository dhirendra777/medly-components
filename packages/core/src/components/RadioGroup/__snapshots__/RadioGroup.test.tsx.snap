// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Group should call onChange prop with selected option 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 0px 8px 0px;
  cursor: default;
}

.c1::after {
  color: red;
  content: ' *';
}

.c6 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 0px 0px 8px;
  cursor: pointer;
}

.c4 {
  grid-area: field;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-left: 0px;
}

.c2 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c11 .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c5 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-columns: max-content minmax(max-content,25%);
  grid-template-areas: 'field label' 'description random';
}

.c5 .c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c0 {
  margin: 8px 0px;
  display: grid;
  grid-template-areas: 'label' 'field' 'description';
}

.c0 .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c10 {
  background-color: #012040;
  width: 0;
  height: 0;
  z-index: 1;
  border-radius: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12:checked ~ .c9 {
  width: calc(75%);
  height: calc(75%);
  -webkit-transition: width 0.2s ease,height 0.2s ease;
  transition: width 0.2s ease,height 0.2s ease;
}

.c8 {
  opacity: 0;
  z-index: 2;
  position: absolute;
  width: 100%;
  height: 100%;
  margin: 0;
  cursor: pointer;
}

.c8:checked ~ .c9 {
  width: calc(75%);
  height: calc(75%);
  -webkit-transition: width 0.2s ease,height 0.2s ease;
  transition: width 0.2s ease,height 0.2s ease;
}

.c7 {
  width: 24px;
  height: 24px;
  position: relative;
  border-radius: 100%;
  border: 1px solid #EEEEEE;
  background-color: #ffffff;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7:focus-within {
  box-shadow: 0 0 0 1pt #012040;
}

<div>
  <div
    class="c0"
    id="Gender-radioGroup"
  >
    <label
      class="c1 c2"
      required=""
    >
      Gender
    </label>
    <div
      class="c3 c4"
    >
      <div
        class="c5"
        id="Female-radio"
      >
        <label
          class="c6 c2"
          for="Female"
        >
          Female
        </label>
        <div
          class="c7"
        >
          <input
            checked=""
            class="c8"
            id="Female"
            name="gender"
            required=""
            type="radio"
            value="female"
          />
          <div
            class="c9 c10"
          />
        </div>
      </div>
      <div
        class="c5"
        id="Male-radio"
      >
        <label
          class="c6 c2"
          for="Male"
        >
          Male
        </label>
        <div
          class="c7"
        >
          <input
            class="c8"
            id="Male"
            name="gender"
            required=""
            type="radio"
            value="male"
          />
          <div
            class="c9 c10"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Radio Group should render correctly 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 8px 0px 0px;
  cursor: default;
}

.c6 {
  font-size: 14px;
  font-weight: 400;
  color: #616161;
  margin: 0px 0px 0px 8px;
  cursor: not-allowed;
}

.c4 {
  grid-area: field;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-left: 0px;
}

.c2 {
  grid-area: label;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  margin: 8px 0px;
  display: grid;
  grid-template-columns: minmax(max-content,25%) minmax(auto,80%);
  grid-template-areas: 'label field' 'random description';
}

.c0 .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c5 {
  margin: 8px 8px 8px 0px;
  display: inline-grid;
  grid-template-columns: max-content minmax(max-content,25%);
  grid-template-areas: 'field label' 'description random';
}

.c5 .c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c10 {
  background-color: #012040;
  width: 0;
  height: 0;
  z-index: 1;
  border-radius: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  opacity: 0;
  z-index: 2;
  position: absolute;
  width: 100%;
  height: 100%;
  margin: 0;
  cursor: not-allowed;
}

.c8:checked ~ .c9 {
  width: calc(75%);
  height: calc(75%);
  -webkit-transition: width 0.2s ease,height 0.2s ease;
  transition: width 0.2s ease,height 0.2s ease;
}

.c7 {
  width: 16px;
  height: 16px;
  position: relative;
  border-radius: 100%;
  border: 1px solid #EEEEEE;
  background-color: #E0E0E0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7:focus-within {
  box-shadow: 0 0 0 1pt #012040;
}

<div>
  <div
    class="c0"
    id="Gender-radioGroup"
  >
    <label
      class="c1 c2"
    >
      Gender
    </label>
    <div
      class="c3 c4"
    >
      <div
        class="c5"
        id="Female-radio"
      >
        <label
          class="c6 c2"
          for="Female"
        >
          Female
        </label>
        <div
          class="c7"
          disabled=""
        >
          <input
            checked=""
            class="c8"
            disabled=""
            id="Female"
            name="gender"
            type="radio"
            value="female"
          />
          <div
            class="c9 c10"
          />
        </div>
      </div>
      <div
        class="c5"
        id="Male-radio"
      >
        <label
          class="c6 c2"
          for="Male"
        >
          Male
        </label>
        <div
          class="c7"
          disabled=""
        >
          <input
            class="c8"
            disabled=""
            id="Male"
            name="gender"
            type="radio"
            value="male"
          />
          <div
            class="c9 c10"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
